src/wallet.js

```diff
@@ -134,27 +135,19 @@ export function WalletBlockNumberProvider(props) {
   const [blockNumber, setBlockNumber] = useState(-1)
 
   useEffect(() => {
-    const sub = WALLET_WEB3.eth.subscribe('newBlockHeaders', (err, block) => {
-      if (!err) {
-        setBlockNumber(block.number)
+    const pollBlockNumber = pollEvery(() => {
+      return {
+        request: () =>
+          WALLET_WEB3.eth
+            .getBlockNumber()
+            .then(latestBlockNumber => latestBlockNumber),
```

Comment:
We can probably remove this line :-)

Reply:
Aah yes, but we will have to make the function async if we do. :) Let's do that as its nicer that way, code-wise.

Reply:
I don’t think it should be necessary? `.getBlockNumber()` returns a promise already like `.then()` do.

Reply:
Yes indeed; Was actually thinking about another piece of code 😅